### ISSUE TITLE
INI lookup fails if default is specified

### ISSUE THREAD

**Issue by SteveTalbot** [ansible/ansible#18369]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

- lookup('ini', ...)

##### ANSIBLE VERSION

```
ansible 2.2.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### OS / ENVIRONMENT

- Running from Ubuntu 16.04
- Managing Ubuntu 16.04

##### SUMMARY

If you specify the `default` field in an INI lookup, the lookup always fails with the error:
```
fatal: [default]: FAILED! => {"failed": true, "msg": "too many values to unpack"}
```

##### STEPS TO REPRODUCE

```
- name: This works
  debug: msg="{{ lookup('ini', 'KeyName section=SectionName file=filename.ini') }}"

- name: This should work, but doesn't
  debug: msg="{{ lookup('ini', 'KeyName section=SectionName file=filename.ini default=XXX') }}"

- name: This doesn't work either
  debug: msg="{{ lookup('ini', 'NonExistentKey section=SectionName file=filename.ini default=XXX') }}"

- name: We don't expect this to work
  debug: msg="{{ lookup('ini', 'NonExistentKey section=SectionName file=filename.ini') }}"
```

##### EXPECTED RESULTS

This is the output from Ansible 2.1.1:
```
TASK [php7-cli : This works] ***************************************************
ok: [default] => {
    "msg": "Value"
}

TASK [php7-cli : This should work, but doesn't] ********************************
ok: [default] => {
    "msg": "Value"
}

TASK [php7-cli : This doesn't work either] *************************************
ok: [default] => {
    "msg": "[]"
}

TASK [php7-cli : We don't expect this to work] *********************************
ok: [default] => {
    "msg": "[]"
}
```

##### ACTUAL RESULTS

```
TASK [php7-cli : This works] ***************************************************
ok: [default] => {
    "msg": "php7.0-bcmath"
}

TASK [php7-cli : This should work, but doesn't] ********************************
fatal: [default]: FAILED! => {
    "failed": true, 
    "msg": "too many values to unpack"
}
```

##### WORKAROUND
Downgrade to Ansible 2.1.1.

**Comment by alikins** [comment]
I can reproduce this with the devel branch

```
[fedora-23:lookup_ini_18369 (master %)]$ cat test.yml 
---
- hosts: localhost
  tasks:
    - name: This works
      debug: msg="{{ lookup('ini', 'KeyName section=SectionName file=filename.ini') }}"

    - name: This should work, but doesn't
      debug: msg="{{ lookup('ini', 'KeyName section=SectionName file=filename.ini default=XXX') }}"

    - name: This doesn't work either
      debug: msg="{{ lookup('ini', 'NonExistentKey section=SectionName file=filename.ini default=XXX') }}"

    - name: We don't expect this to work
      debug: msg="{{ lookup('ini', 'NonExistentKey section=SectionName file=filename.ini') }}"
[fedora-23:lookup_ini_18369 (master %)]$ cat filename.ini 
[SectionName]
file=ddddfilename.ini
[fedora-23:lookup_ini_18369 (master %)]$ ansible-playbook -vvvv test.yml 
Using /home/adrian/.ansible.cfg as config file
 [WARNING]: provided hosts list is empty, only localhost is available

Loading callback plugin default of type stdout, v2.0 from /home/adrian/src/ansible/lib/ansible/plugins/callback/__init__.pyc

PLAYBOOK: test.yml ********************************************************************************************************************************************************************************************************************
1 plays in test.yml

PLAY [localhost] **********************************************************************************************************************************************************************************************************************

TASK [setup] **************************************************************************************************************************************************************************************************************************
Using module file /home/adrian/src/ansible/lib/ansible/modules/core/system/setup.py
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: adrian
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1478530448.12-177352847997456 `" && echo ansible-tmp-1478530448.12-177352847997456="` echo $HOME/.ansible/tmp/ansible-tmp-1478530448.12-177352847997456 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpBFjlcU TO /home/adrian/.ansible/tmp/ansible-tmp-1478530448.12-177352847997456/setup.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /home/adrian/.ansible/tmp/ansible-tmp-1478530448.12-177352847997456/ /home/adrian/.ansible/tmp/ansible-tmp-1478530448.12-177352847997456/setup.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python /home/adrian/.ansible/tmp/ansible-tmp-1478530448.12-177352847997456/setup.py; rm -rf "/home/adrian/.ansible/tmp/ansible-tmp-1478530448.12-177352847997456/" > /dev/null 2>&1 && sleep 0'
ok: [localhost]

TASK [This works] *********************************************************************************************************************************************************************************************************************
task path: /home/adrian/src/ansible-bug-repro/lookup_ini_18369/test.yml:4
ok: [localhost] => {
    "msg": []
}

TASK [This should work, but doesn't] **************************************************************************************************************************************************************************************************
task path: /home/adrian/src/ansible-bug-repro/lookup_ini_18369/test.yml:7
fatal: [localhost]: FAILED! => {
    "failed": true, 
    "msg": "too many values to unpack"
}
    to retry, use: --limit @/home/adrian/src/ansible-bug-repro/lookup_ini_18369/test.retry

PLAY RECAP ****************************************************************************************************************************************************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=1   

[fedora-23:lookup_ini_18369 (master %)]$ ansible --version
ansible 2.3.0 (devel 5aaf1d1a15) last updated 2016/11/07 09:32:34 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 2584fca0ae) last updated 2016/11/04 13:15:56 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD a1dcbf9ce5) last updated 2016/11/04 13:15:56 (GMT -400)
  config file = /home/adrian/.ansible.cfg
  configured module search path = Default w/o overrides


```

**Comment by alikins** [comment]
bisect points this to 4ba60d00c8d7e62912a37ec24f90f6e5d0770c4d

```
commit 4ba60d00c8d7e62912a37ec24f90f6e5d0770c4d
Author: jctanner <tanner.jc@gmail.com>
Date:   Tue Jun 28 13:24:57 2016 -0400

    Refactor the parameter splitting in ini lookup to handle more path formats (#16477)

    Refactor the parameter splitting in ini lookup to handle more path formats.

    Fixes #16468
```

**Comment by alikins** [comment]
reproducer and bisect script at https://github.com/alikins/ansible-bug-repro/tree/master/lookup_ini_18369
